moveit_ros/moveit_servo/src/servo.cpp

```diff
@@ -374,10 +376,7 @@ Eigen::VectorXd Servo::jointDeltaFromCommand(const ServoInput& command, moveit::
 KinematicState Servo::getNextJointState(const ServoInput& command)
 {
   // Set status to clear
-  servo_status_ = StatusCode::NO_WARNING;
-
-  // Update the parameters
-  updateParams();
+  setStatus(StatusCode::NO_WARNING);
```

Comment:
I think this should be done outside of this function, for readability. It's kind of hidden away here in an unrelated function.

Reply:
This is the only place we can set it automatically.
Otherwise user would have to manually call it.

Reply:
If you have the function either return the status or accept it as a mutable input arg, this could be done!

Reply:
Reading through the code again, I realize that additionally to everything already said, using the `setStatus()` function the way we use it now is why you had to make all those other member functions non-`const`. So another data point for figuring this out in a more functional way.